File: class_primer/class_primer__heros.cs
using System.Linq;
using static System.Console;

internal class Program
{
    class Hero
    {
        int Agi { get; set; }
        int Atk { get; set; }
        int Def { get; set; }
        int Hlt { get; set; }
        int Int { get; set; }
        int Luc { get; set; }
        int Lvl { get; set; }

        internal Hero( int[] a )
        {
            Lvl = a[0];
            Hlt = a[1];
            Atk = a[2];
            Def = a[3];
            Agi = a[4];
            Int = a[5];
            Luc = a[6];
        }

        internal void LevelUp( int[] a )
        {
            Lvl++;
            Hlt += a[0];
            Atk += a[1];
            Def += a[2];
            Agi += a[3];
            Int += a[4];
            Luc += a[5];
            return;
        }

        internal void MuscleTraining( int[] a )
        {
            Hlt += a[0];
            Atk += a[1];
            return;
        }

        internal void Running( int[] a )
        {
            Def += a[0];
            Agi += a[1];
            return;
        }

        internal void Study( int[] c )
        {
            Int += c[0];
            return;
        }

        internal void Pray( int[] f )
        {
            Luc += f[0];
            return;
        }

        internal void Print()
        {
            WriteLine( $"{Lvl} {Hlt} {Atk} {Def} {Agi} {Int} {Luc}" );
            return;
        }
    }

    static void Main()
    {
        int[] n = ReadLine().Split().Select( int.Parse ).ToArray();
        Hero[] h = new Hero[n[0]];

        foreach( int i in Enumerable.Range( 0, n[0] ) ) {
            h[i] = new Hero( ReadLine().Split().Select( int.Parse ).ToArray() );
        }
        foreach( int _ in Enumerable.Range( 0, n[1] ) ) {
            string[] s = ReadLine().Split();
            int t = int.Parse( s[0] ) - 1;
            int[] a = s.Skip( 2 ).Select( int.Parse ).ToArray();

            switch( s[1] ) {
            case "levelup":
                h[t].LevelUp( a );
                break;
            case "muscle_training":
                h[t].MuscleTraining( a );
                break;
            case "running":
                h[t].Running( a );
                break;
            case "study":
                h[t].Study( a );
                break;
            case "pray":
                h[t].Pray( a );
                break;
            }
        }
        SetOut( new System.IO.StreamWriter( OpenStandardOutput() ) { AutoFlush = false } );
        foreach( Hero i in h ) {
            i.Print();
        }
        Out.Flush();
        return;
    }
}

File: class_primer/class_primer__heros.py
class hero:
    def __init__(self, a):
        self.lvl = a[0]
        self.hlt = a[1]
        self.atk = a[2]
        self.dfd = a[3]
        self.agi = a[4]
        self.int = a[5]
        self.luk = a[6]

    def levelUp(self, a):
        self.lvl += 1
        self.hlt += int(a[2])
        self.atk += int(a[3])
        self.dfd += int(a[4])
        self.agi += int(a[5])
        self.int += int(a[6])
        self.luk += int(a[7])

    def muscleTraining(self, a):
        self.hlt += int(a[2])
        self.atk += int(a[3])

    def running(self, a):
        self.dfd += int(a[2])
        self.agi += int(a[3])

    def study(self, i):
        self.int += int(i[2])

    def pray(self, l):
        self.luk += int(l[2])

    def write(self):
        print(f"{self.lvl} {self.hlt} {self.atk} {self.dfd} {self.agi} {self.int} {self.luk}")

n, k = list(map(int, input().split()))
l = []

for _ in range(n):
    l.append(hero(list(map(int, input().split()))))
for _ in range(k):
    s = input().split()
    i = int(s[0]) - 1

    if s[1] == "levelup":
        l[i].levelUp(s)
    elif s[1] == "muscle_training":
        l[i].muscleTraining(s)
    elif s[1] == "running":
        l[i].running(s)
    elif s[1] == "study":
        l[i].study(s)
    elif s[1] == "pray":
        l[i].pray(s)
for i in l:
    i.write()
